### PR TITLE
Fix i18n fallback configuration conflicts with environment configurations

### PR DESCRIPTION
**Pull request by akihikodaki**
This fixes a bug reported at https://github.com/tootsuite/mastodon/pull/6841#issuecomment-374607268.

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -56,7 +56,7 @@
 
   # Enable locale fallbacks for I18n (makes lookups for any locale fall back to
   # the I18n.default_locale when a translation cannot be found).
```
Comment (ineffyble) [inline config/environments/production.rb]: Is this comment still accurate? Seems like it should say something more like "make lookups for any locale fall back to 'en'"?

Comment (akihikodaki) [reply config/environments/production.rb]: No. Addressed in https://github.com/tootsuite/mastodon/pull/6843/commits/7820893aca79826b19c51da8dc834b13278db0b7. Thanks for pointing out this.

**Comment by ineffyble** [comment]
To clarify how this works for my own benefit:
- Instance owner sets a default locale via environment variable
- This default locale is used for all the localised strings shown within Mastodon
- If a string is not available in this locale, Mastodon will instead use the English version of the string

In addition:
- If the default locale is not provided, or is unavailable or invalid, it is set to English

Does that all sound right?

**Comment by akihikodaki** [comment]
Yes. And it has been the expected behavior, so it is a _bug_ if it doesn't work as you stated. (That's why this pull request is labelled _bug_ here.)

**Review by nightpool — APPROVED** [review]
i just touched this code recently for cybrespace-specific locale changes, so i can confirm this is the correct change to make. thanks aki!

**Comment by nightpool** [comment]
(to be clear, the bug here is that we were setting `.fallbacks = [:en]` in application.rb, but it was getting overwritten later by the default `.fallbacks = true` in production.rb.

**Review by ineffyble — APPROVED** [review]

**Pull request merged**